int32 AnimationType
int32 AnimationID